python_ta/transforms/type_inference_visitor.py

```diff
@@ -102,28 +102,23 @@ def visit_tuple(self, node):
 
     def visit_list(self, node):
         # node_types contains types of elements inside list.
-        node_types = {node_child.type_constraints.type for node_child in node.elts}
-        # If list has more than one type, just set node.type_constraints to List.
-        # If list has only one type T, set the node.type_constraints to be List[T].
-        if len(node_types) == 1:
-            # node_types.pop() returns the only element in the set, which is a
-            # type object.
-            node.type_constraints = TypeInfo(List[node_types.pop()])
-        else:
-            node.type_constraints = TypeInfo(List[Any])
+        node_type = node.elts[0].type_constraints.type
```

Comment:
This is good except that it will fail on an empty list. This is a bit trickier, so for now you can do what the old code did and set the type constraint to `List[Any]` in that case.

Similar for the other collection types.